packages/react-components/src/components/ConfigurationUserPreferencesStep.tsx

```diff
@@ -1,47 +1,191 @@
-import React, { ChangeEvent, FC } from 'react';
-import Label from './Label';
+import React, { FC, useEffect, useState } from 'react';
+import {
+    ConfigurationStep,
+    Severity,
+    SourceType,
+    SourceTypeCategory,
+    Tenant,
+} from '../models/flare';
 import Button from './Button';
-import { Tenant } from '../models/flare';
+import Label from './Label';
 import Select from './Select';
 
+import { appName, DEFAULT_FILTER_VALUE } from '../models/constants';
+import {
+    fetchAvailableIndexNames,
+    fetchCurrentIndexName,
+    fetchFiltersSeverities,
+    fetchFiltersSourceTypes,
+    fetchIngestMetadataOnly,
+    fetchSeveritiesFilter,
+    fetchSourceTypesFilter,
+    fetchTenantId,
+    fetchUserTenants,
+    saveConfiguration,
+} from '../utils/setupConfiguration';
 import './ConfigurationGlobalStep.css';
 import './ConfigurationUserPreferencesStep.css';
+import SeverityOptions from './SeverityOptions';
+import SourceTypeCategoryOptions from './SourceTypeCategoryOptions';
 import Switch from './Switch';
+import { ToastKeys, toastManager } from './ToastManager';
+import Tooltip from './Tooltip';
 
 const ConfigurationUserPreferencesStep: FC<{
     show: boolean;
-    tenants: Tenant[];
-    selectedTenantId: number;
-    indexNames: string[];
-    selectedIndexName: string;
-    isLoading: boolean;
-    isIngestingMetadataOnly: boolean;
+    configurationStep: ConfigurationStep;
+    apiKey: string;
     onNavigateBackClick: () => void;
-    onSubmitUserPreferencesClick: () => void;
-    onTenantIdChange: (e: ChangeEvent) => void;
-    onIndexNameChange: (e: ChangeEvent) => void;
-    onIngestingMetadataChange: (e: ChangeEvent) => void;
-}> = ({
-    show,
-    tenants,
-    selectedTenantId,
-    indexNames,
-    selectedIndexName,
-    isLoading,
-    isIngestingMetadataOnly,
-    onNavigateBackClick,
-    onSubmitUserPreferencesClick,
-    onTenantIdChange,
-    onIndexNameChange,
-    onIngestingMetadataChange,
-}) => {
+    onUserPreferencesSaved: () => void;
+}> = ({ show, configurationStep, apiKey, onNavigateBackClick, onUserPreferencesSaved }) => {
+    const [tenantId, setTenantId] = useState(-1);
```

Comment:
What does the -1 do here? Google is failing me. Rather, I'm probably failing Google.